app/src/test/java/org/mozilla/tv/firefox/framework/FrameworkRepoTest.kt

```diff
@@ -0,0 +1,65 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.tv.firefox.framework
+
+import android.arch.core.executor.testing.InstantTaskExecutorRule
+import android.view.accessibility.AccessibilityManager
+import android.view.accessibility.AccessibilityManager.TouchExplorationStateChangeListener
+import org.junit.Assert.assertFalse
+import org.junit.Assert.assertTrue
+import org.junit.Before
+import org.junit.Rule
+import org.junit.Test
+import org.junit.runner.RunWith
+import org.mockito.Mockito.`when`
+import org.mockito.Mockito.any
+import org.mockito.Mockito.mock
+import org.mockito.Mockito.times
+import org.mockito.Mockito.verify
+import org.robolectric.RobolectricTestRunner
+
+@RunWith(RobolectricTestRunner::class)
+class FrameworkRepoTest {
+
+    @get:Rule val instantTaskRule = InstantTaskExecutorRule() // necessary for LiveData tests.
+
+    private lateinit var repo: FrameworkRepo
+
+    @Before
+    fun setUp() {
+        repo = FrameworkRepo()
+    }
+
+    @Test
+    fun `WHEN init is called THEN a touch exploration state change listener is added`() {
+        val a11yManager = mock(AccessibilityManager::class.java)
+        repo.init(a11yManager)
+        verify(a11yManager, times(1)).addTouchExplorationStateChangeListener(any(TouchExplorationStateChangeListener::class.java))
+    }
```

Comment:
> but these tests seem to be very much concerned with implementation details.

tbh, I don't have a good idea of how to conceptualize if a test is too concerned with implementation details. I guess I need to do more reading.

> but I'm not sure they're particularly helpful in this case. This strategy can also make changing implementation a more laborious process.

I really want to test that whenever the dependency pushes out an accessibility state change, the repo updates its internal state accordingly: i.e. is it reacting? This is an important test: if someone modifies this code, we want to ensure the framework will still correctly update the repo. This test would require an implementation detail – what is the dependency? – but I don't think it's possible to avoid that. I'll try to rewrite these tests to be more general but it'll require afaict, it'd require a more comprehensive shadow/mock of the AccessibilityManager.

> Looking at the class being tested, ...

I'll get back to the rest of the comment after I've addressed the tests.

Reply:
> Will a value be immediately available after init?

Incidentally added a test for this.

> Can init be called twice? What will happen if it is?

Added this functionality and added a test.